pkg/network/protocols/core/events.go

```diff
@@ -3,102 +3,48 @@ package core
 import (
 	"github.com/iotaledger/hive.go/crypto/identity"
 	"github.com/iotaledger/hive.go/runtime/event"
-	"github.com/iotaledger/iota-core/pkg/commitment"
-	"github.com/iotaledger/iota-core/pkg/models"
-	"github.com/iotaledger/iota-core/pkg/slot"
+	iotago "github.com/iotaledger/iota.go/v4"
 )
 
-// region Events ///////////////////////////////////////////////////////////////////////////////////////////////////////
-
 type Events struct {
-	BlockReceived                 *event.Event1[*BlockReceivedEvent]
-	BlockRequestReceived          *event.Event1[*BlockRequestReceivedEvent]
-	SlotCommitmentReceived        *event.Event1[*SlotCommitmentReceivedEvent]
-	SlotCommitmentRequestReceived *event.Event1[*SlotCommitmentRequestReceivedEvent]
+	BlockReceived                 *event.Event2[*Block, identity.ID]
+	BlockRequestReceived          *event.Event2[iotago.BlockID, identity.ID]
+	SlotCommitmentReceived        *event.Event2[*iotago.Commitment, identity.ID]
+	SlotCommitmentRequestReceived *event.Event2[iotago.CommitmentID, identity.ID]
 	AttestationsReceived          *event.Event1[*AttestationsReceivedEvent]
 	AttestationsRequestReceived   *event.Event1[*AttestationsRequestReceivedEvent]
-	Error                         *event.Event1[*ErrorEvent]
+	Error                         *event.Event2[error, identity.ID]
 
 	event.Group[Events, *Events]
 }
 
 // NewEvents contains the constructor of the Events object (it is generated by a generic factory).
 var NewEvents = event.CreateGroupConstructor(func() (newEvents *Events) {
 	return &Events{
-		BlockReceived:                 event.New1[*BlockReceivedEvent](),
-		BlockRequestReceived:          event.New1[*BlockRequestReceivedEvent](),
-		SlotCommitmentReceived:        event.New1[*SlotCommitmentReceivedEvent](),
-		SlotCommitmentRequestReceived: event.New1[*SlotCommitmentRequestReceivedEvent](),
+		BlockReceived:                 event.New2[*Block, identity.ID](),
+		BlockRequestReceived:          event.New2[iotago.BlockID, identity.ID](),
+		SlotCommitmentReceived:        event.New2[*iotago.Commitment, identity.ID](),
+		SlotCommitmentRequestReceived: event.New2[iotago.CommitmentID, identity.ID](),
 		AttestationsReceived:          event.New1[*AttestationsReceivedEvent](),
 		AttestationsRequestReceived:   event.New1[*AttestationsRequestReceivedEvent](),
```

Comment:
Why not change these to `event.New3`?

Reply:
I wanted to wait until we can use the attestations properly before we change this signature